services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/AsyncBasicCrudTest.java

```diff
@@ -170,27 +168,43 @@ public int hashCode() {
 
     private static final TableSchema<Record> TABLE_SCHEMA =
         StaticTableSchema.builder(Record.class)
-                   .newItemSupplier(Record::new)
-                   .attributes(
-                       attribute("id", TypeToken.of(String.class), Record::getId, Record::setId).as(primaryPartitionKey()),
-                       attribute("sort", TypeToken.of(String.class), Record::getSort, Record::setSort).as(primarySortKey()),
-                       // This is a DynamoDb reserved word, forces testing of AttributeNames
-                       attribute("attribute", TypeToken.of(String.class), Record::getAttribute, Record::setAttribute),
-                       // Using tricky characters to force scrubbing of attributeName tokens
-                       attribute("*attribute2*", TypeToken.of(String.class), Record::getAttribute2, Record::setAttribute2)
-                           .as(secondaryPartitionKey("gsi_1")),
-                       attribute("attribute3", TypeToken.of(String.class), Record::getAttribute3, Record::setAttribute3)
-                           .as(secondarySortKey("gsi_1")))
-                   .build();
+                         .newItemSupplier(Record::new)
+                         .addAttribute(String.class, a -> a.name("id")
+                                                           .getter(Record::getId)
+                                                           .setter(Record::setId)
+                                                           .tags(primaryPartitionKey()))
+                         .addAttribute(String.class, a -> a.name("sort")
+                                                           .getter(Record::getSort)
+                                                           .setter(Record::setSort)
+                                                           .tags(primarySortKey()))
+                         .addAttribute(String.class, a -> a.name("attribute")
+                                                           .getter(Record::getAttribute)
+                                                           .setter(Record::setAttribute))
+                         .addAttribute(String.class, a -> a.name("attribute2*")
```

Comment:
I have noticed that in many test files, we're defining but not using the secondary index. Should we consider adding more tests to target that and cover more behavior?

Reply:
Query and scan are the only two operations currently supported by secondary indices and we have good test coverage on those. These tests share a lot of the POJOs so in a way what is being tested here is that non-index operations work fine when indexes are present.

Reply:
ok